### PR TITLE
Reimplement player as characterbody2d

### PR DESCRIPTION
**Pull request by JojSh**
## Description

Reimplement player as characterbody2d:
- [Make Player a child of Main (instead of TileMap)](https://github.com/JojSh/Choppa/commit/69eb0ec801c0a32db9d48a172ba7105c30e0156c)
- [Paint the obstacles in TileMap](https://github.com/JojSh/Choppa/commit/d4fc9594bf372a1e782c44f6d468277c1fccd99d)
- [Change something to do with keycodes I think (not sure)](https://github.com/JojSh/Choppa/commit/c14ec36372a3f0168c548923b0bb0aefd214ef9d)
- [Change Player node from Area2D to CharacterBody2D](https://github.com/JojSh/Choppa/commit/fd511fa74307995c10043ee3b0a637e03747bcd0)
- [Move Player with physics instead of position change](https://github.com/JojSh/Choppa/commit/9111900e73420c6e06a2f1e4220178daf29f4d51)

## Video
Before:
https://github.com/JojSh/Choppa/assets/16098730/82b738ba-969e-4298-87e8-453341fcb187

After:
https://github.com/JojSh/Choppa/assets/16098730/cc1bdc38-a07b-4841-b48b-d9b4479467ba

### CODE REVIEW

File: Player.gd [hunk 1]
```diff
@@ -1,13 +1,25 @@
-extends Area2D
+extends CharacterBody2D
 
 var move_speed_px : int = 50
 
-func _process(delta):
+func _physics_process (delta):
+	velocity = Vector2()
```
Comment (Teeohbee) [inline Player.gd]: I'm assuming these are equivalent but is this a little clearer?
```suggestion
	velocity = Vector2.ZERO
```

File: Player.gd [hunk 2]
```diff
@@ -1,13 +1,25 @@
...
+		velocity.x -= 1
+	if Input.is_action_pressed("move_right"):
+		velocity.x += 1
+	
+	velocity = velocity.normalized() * move_speed_px
+	move_and_slide()
 	handle_animation(direction)
 
-func handle_animation(direction):
+func handle_animation (direction):
```
Comment (Teeohbee) [inline Player.gd:22]: Unnecessary white space 
```suggestion
func handle_animation(direction):
```

File: main.tscn [hunk 3]
```diff
@@ -3244,5 +3285,5 @@ layer_1/y_sort_origin = 0
...
 
-[node name="Player" parent="TileMap" instance=ExtResource("2_rp60p")]
+[node name="Player" parent="." instance=ExtResource("2_rp60p")]
```
Comment (Teeohbee) [inline main.tscn:3288]: Interesting, did we have the player as a child of the tile map?

**Review by Teeohbee — APPROVED** [review]
LGTM, just some suggestions and comments.

File: main.tscn (hunk 3 above)
Comment (JojSh) [reply main.tscn:3288]: Yes. Accidentally I think!

File: Player.gd (hunk 2 above)
Comment (JojSh) [reply Player.gd:22]: Maybe later 😉

**Pull request merged**